core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/elasticapi/ElasticExtensions.kt

```diff
@@ -153,28 +153,19 @@ suspend fun <C : ElasticsearchClient, T> C.suspendUntil(block: C.(ActionListener
 /**
  * Store a [ThreadContext] and restore a [ThreadContext] when the coroutine resumes on a different thread.
  *
- * @param threadContext - The context to store when switching to a coroutine.
- * @param initialContext - The old context to restore upon completion of the coroutine.
+ * @param threadContext - a [ThreadContext] instance
  */
-class ElasticThreadContextElement(
-    private val threadContext: ThreadContext
-) : ThreadContextElement<StoredContext> {
+class ElasticThreadContextElement(private val threadContext: ThreadContext) : ThreadContextElement<Unit> {
 
     companion object Key : CoroutineContext.Key<ElasticThreadContextElement>
-    private var initialContext: StoredContext? = threadContext.newStoredContext(true)
+    private var initialContext: StoredContext = threadContext.newStoredContext(true)
```

Comment:
Can this be `val` now that we don't re-assign this variable?

Reply:
We reassign it on line 167
```
        initialContext = threadContext.stashContext()
```